test/e2e/data/infrastructure-docker/bases/mhc.yaml

```diff
@@ -0,0 +1,18 @@
+---
+# MachineHealthCheck object with
+# - a selector that targets all the machines with label nodepool=pool1
+# - unhealthyConditions triggering remediation after 30s the node is up (because it is testing a condition that does not exists)
+apiVersion: cluster.x-k8s.io/v1alpha3
+kind: MachineHealthCheck
+metadata:
+  name: "${CLUSTER_NAME}-mhc-0"
+spec:
+  clusterName: "${CLUSTER_NAME}"
+  maxUnhealthy: 100%
+  selector:
+    matchLabels:
+      nodepool: "pool1"
+  unhealthyConditions:
+    - type: E2ENodeUnhealthy
+      status: "True"
+      timeout: 30s
```

Comment:
Can we revisit this? I understand the goal here is to make MHC fail and remediate after 30s, but if we could actually flip this to control when this happens and reduce the timeout, that would probably be better?

For example, our tests could be a little bit more controlled if we apply the unhealthy condition, wait for the timeout, and ultimately check that health check or remediation happened.

Reply:
I agree there could be cleaner solutions, like patching the node with our own conditions (and if I remember well @sedefsavas explored some of them).
However for the sake of testing, also the current solution is achieving the goal of triggering remediation in a consistent way and this test is pretty stable...

Reply:
Create a follow up issue perhaps?

Reply:
Yes a follow-up is fine, I wanted to discuss it though given that this PR brought it up

Reply:
https://github.com/kubernetes-sigs/cluster-api/issues/3656